timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/NamedMinTimestampTrackerImpl.java

```diff
@@ -46,6 +64,7 @@ public synchronized void lock(long timestamp, UUID requestId) {
                     SafeArg.of("requestId", requestId),
```

Comment:
One thing to note is: the code path is now different than before as this impacts the immutable timestamp. I can special case.

Of course, with all concurrency comments on this PR, I may be too paranoid here